### PR TITLE
e2e tests for SslPolicy

### PR DESCRIPTION
**Pull request by spencerhance**
Also includes some frontendconfig helpers for e2e tests

### CODE REVIEW

File: pkg/fuzz/helpers.go [hunk 1]
```diff
@@ -411,3 +421,41 @@ func (b *BackendConfigBuilder) SetHealthCheckPath(path string) *BackendConfigBui
...
+// purposes.
+//
+//   frontendConfig := NewFrontendConfigBuilder("ns1", "name1").Build()
+type FrontendConfigBuilder struct {
+	frontendConfig *frontendconfig.FrontendConfig
+}
+
+// NewFrontendConfigBuilder instantiates a new FrontendConfig.
+func NewFrontendConfigBuilder(ns, name string) *FrontendConfigBuilder {
+	var frontendConfig = &frontendconfig.FrontendConfig{
```
Comment (rramkumar1) [inline pkg/fuzz/helpers.go:435]: can just call this fc to reduce verbosity.

File: cmd/e2e-test/sslpolicy_test.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	port80 := intstr.FromInt(80)
+
+	for _, tc := range []struct {
+		desc       string
+		ingBuilder *fuzz.IngressBuilder
+		hosts      []string
+		policy     *compute.SslPolicy
+	}{
+		{
+			desc: "Set and Overwrite SslPolicy",
```
Comment (rramkumar1) [inline cmd/e2e-test/sslpolicy_test.go:47]: What about this test is overwriting? Seems like you just set it then check.

File: pkg/fuzz/helpers.go (hunk 1 above)
Comment (spencerhance) [reply pkg/fuzz/helpers.go:435]: Done

File: cmd/e2e-test/sslpolicy_test.go (hunk 2 above)
Comment (spencerhance) [reply cmd/e2e-test/sslpolicy_test.go:47]: Done

File: cmd/e2e-test/sslpolicy_test.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (bowei) [inline cmd/e2e-test/sslpolicy_test.go:2]: 2020

File: cmd/e2e-test/sslpolicy_test.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+			policy: &compute.SslPolicy{Name: "e2e-ssl-policy", MinTlsVersion: "TLS_1_0", Profile: "COMPATIBLE"},
+		},
+	} {
+		tc := tc // Capture tc as we are running this in parallel.
+		Framework.RunWithSandbox(tc.desc, t, func(t *testing.T, s *e2e.Sandbox) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			// Setup Certificates
```
Comment (bowei) [inline cmd/e2e-test/sslpolicy_test.go:61]: should we do this outsdie the loop?

File: cmd/e2e-test/sslpolicy_test.go (hunk 3 above)
Comment (spencerhance) [reply cmd/e2e-test/sslpolicy_test.go]: Done

File: cmd/e2e-test/sslpolicy_test.go (hunk 4 above)
Comment (spencerhance) [reply cmd/e2e-test/sslpolicy_test.go:59]: Done

Comment (bowei) [reply cmd/e2e-test/sslpolicy_test.go:59]: I meant outside the test loop, this will create and delete certs for each test case.

Comment (spencerhance) [reply cmd/e2e-test/sslpolicy_test.go]: Ah I misunderstood, fixed it now

**Comment by spencerhance** [comment]
@bowei @rramkumar1 Can we merge this?

**Pull request merged**